### PR TITLE
Create NuGet packages for MSBuild.exe and built-in targets files, and various other changes

### PR DESCRIPTION
**Pull request by dsplaisted**
- Create `MSBuild` NuGet package with MSBuild.exe and `Microsoft.Build.Targets` with built-in targets/props/etc files
- Fix incremental build of _RestoreBuildToolsPackagesConfig target when target-specific packages file doesn't exist (ie on Windows)
- Port CreateVisualBasicManifestResourceName task to .NET Core
- Update Debugging target in VS to use DeploymentDir (so F5 in VS works)
- Update to NuGet 3.3, use version-specific semaphore file so that when version is updated it will be re-downloaded (overlaps with #383)
- Do NuGet restore using project.json files instead of .sln file (which has issues when building on .NET Core with full framework NuGet.exe)

### CODE REVIEW

File: src/.nuget/packageLoad.targets [hunk 1]
```diff
@@ -74,7 +77,15 @@
              Condition="!Exists('$(NuGetDir)')" />
 
     <!-- Download NuGet.exe -->
-    <Exec Command="$(DownloadCommand)" Condition="!Exists('$(NuGetToolPath)')" />
+    <Exec Command="$(DownloadCommand)" Condition="!Exists('$(NugetExeSemaphore)')" />
+
+    <Touch Files="$(NuGetToolPath)" />
```
Comment (cdmihai) [inline src/.nuget/packageLoad.targets]: Why do you also touch Nuget.exe?

File: src/.nuget/packageLoad.targets [hunk 2]
```diff
@@ -107,30 +119,10 @@
...
-           AlwaysCreate="true"
-           ForceTouch="true" />
-  </Target>
-    
   <Target Name="_RestoreDnuProjectJsonReferences"
           Inputs="@(ProjectJsonFile)"
           Outputs="%(ProjectJsonFile.RootDir)%(ProjectJsonFile.Directory)project.lock.json"
-          AfterTargets="_RestoreGlobalReferences"
-          Condition="('$(SolutionFile)'=='' or !Exists('$(SolutionFile)')) and '$(OS)'!='Windows_NT'"  >
+          AfterTargets="_RestoreGlobalReferences" >
```
Comment (cdmihai) [inline src/.nuget/packageLoad.targets]: This makes the crossplatform build more unified. Cool.

**Comment by cdmihai** [comment]
@dotnet-bot test this please

**Comment by rainersigwald** [comment]
@dotnet-bot test this please

**Comment by cdmihai** [comment]
@dotnet-bot test this please

**Comment by dsplaisted** [comment]
@cdmihai @rainersigwald This works on my machine (TM) but the CI is failing.  Is the CI currently working?

**Comment by rainersigwald** [comment]
@dsplaisted It is for Windows.  I'm trying to set up a clean VM to see if I can repro the failures.

**Comment by rainersigwald** [comment]
My psychic debugging powers suggest that you're building from within Powershell, aren't you?

Can you cherry-pick eb6f9fb into this?  That made it work on my clean VM, and I'm trying it as #393.  The issue was that `$(HOME)` was coming out as an empty string.

**Comment by dsplaisted** [comment]
These changes have been merged in #393

**Pull request merged**